diff -Nur Electrum-2.6.4.ori/electrum Electrum-2.6.4/electrum
--- Electrum-2.6.4.ori/electrum	2016-03-08 03:51:00.000000000 -0500
+++ Electrum-2.6.4/electrum	2016-04-25 20:10:41.856607707 -0400
@@ -106,8 +106,6 @@
     cmdname = config.get('cmd')
 
     storage = WalletStorage(config.get_wallet_path())
-    if storage.file_exists:
-        sys.exit("Error: Remove the existing wallet first!")
 
     def password_dialog():
         return prompt_password("Password (hit return if you do not wish to encrypt your wallet):")
@@ -144,6 +142,7 @@
         print_msg("Please keep it in a safe place; if you lose it, you will not be able to restore your wallet.")
 
     elif cmdname == 'deseed':
+        wallet = Wallet(storage)
         if not wallet.seed:
             print_msg("Error: This wallet has no seed")
         else:
@@ -153,6 +152,8 @@
                 wallet.storage.path = ns
                 wallet.seed = ''
                 wallet.storage.put('seed', '')
+                wallet.storage.put('master_private_keys','')
+                wallet.storage.put('wallet_type','xpub')
                 wallet.use_encryption = False
                 wallet.storage.put('use_encryption', wallet.use_encryption)
                 for k in wallet.imported_keys.keys():
diff -Nur Electrum-2.6.4.ori/gui/qt/installwizard.py Electrum-2.6.4/gui/qt/installwizard.py
--- Electrum-2.6.4.ori/gui/qt/installwizard.py	2016-04-05 04:47:36.000000000 -0400
+++ Electrum-2.6.4/gui/qt/installwizard.py	2016-04-23 22:48:08.712479755 -0400
@@ -249,6 +249,7 @@
         t = threading.Thread(target = task)
         t.start()
         self.please_wait.setText(MSG_GENERATING_WAIT)
+        t.join()
         self.refresh_gui()
 
     def query_create_or_restore(self, wallet_kinds):
diff -Nur Electrum-2.6.4.ori/lib/wizard.py Electrum-2.6.4/lib/wizard.py
--- Electrum-2.6.4.ori/lib/wizard.py	2016-02-24 04:21:16.000000000 -0500
+++ Electrum-2.6.4/lib/wizard.py	2016-04-23 22:47:36.831756320 -0400
@@ -48,9 +48,6 @@
     user_actions = ('create', 'restore')
     wallet_kinds = [
         ('standard',  _("Standard wallet")),
-        ('twofactor', _("Wallet with two-factor authentication")),
-        ('multisig',  _("Multi-signature wallet")),
-        ('hardware',  _("Hardware wallet")),
     ]
 
     # Derived classes must set:
@@ -167,8 +164,6 @@
             # Show network dialog if config does not exist
             if self.config.get('auto_connect') is None:
                 self.choose_server(network)
-        else:
-            self.show_warning(_('You are offline'))
 
         if need_sync:
             self.create_addresses(wallet)
